### PR TITLE
COM parameter ownership doc

### PR DESCRIPTION
**Pull request by jtschuster**
## Summary

Creates a public document for detailing value ownership transfer across COM method calls.

The formatting style and wording needs to be changed and more examples need to be added, but I want to start with feedback on if this is the right way to think about COM ownership and explain it to users.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/standard/native-interop/com-lifetimes-and-ownership-examples.md](https://github.com/dotnet/docs/blob/ede581ceefc1bb3ea63e96d8d77c2fc02cf19e26/docs/standard/native-interop/com-lifetimes-and-ownership-examples.md) | [docs/standard/native-interop/com-lifetimes-and-ownership-examples](https://review.learn.microsoft.com/en-us/dotnet/standard/native-interop/com-lifetimes-and-ownership-examples?branch=pr-en-us-36095) |
| [docs/standard/native-interop/com-parameter-ownership.md](https://github.com/dotnet/docs/blob/ede581ceefc1bb3ea63e96d8d77c2fc02cf19e26/docs/standard/native-interop/com-parameter-ownership.md) | [docs/standard/native-interop/com-parameter-ownership](https://review.learn.microsoft.com/en-us/dotnet/standard/native-interop/com-parameter-ownership?branch=pr-en-us-36095) |
| [docs/standard/native-interop/cominterop.md](https://github.com/dotnet/docs/blob/ede581ceefc1bb3ea63e96d8d77c2fc02cf19e26/docs/standard/native-interop/cominterop.md) | [COM Interop in .NET](https://review.learn.microsoft.com/en-us/dotnet/standard/native-interop/cominterop?branch=pr-en-us-36095) |

### CODE REVIEW

File: docs/standard/native-interop/com-lifetimes-and-ownership-examples.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+int Method1(int x);
+int Method2(float y);
+int Method3(PrimitiveStruct s);
+struct PrimitiveStruct
+{
+    int a;
+    float b;
+}
+```
+```cpp
```
Comment (BillWagner) [inline docs/standard/native-interop/com-lifetimes-and-ownership-examples.md:23]: lint fix:

```suggestion

```cpp
```

File: docs/standard/native-interop/com-lifetimes-and-ownership-examples.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+int Method1(int x);
+int Method2(float y);
+int Method3(PrimitiveStruct s);
+struct PrimitiveStruct
+{
+    int a;
+    float b;
+}
+```
+
```
Comment (BillWagner) [inline docs/standard/native-interop/com-lifetimes-and-ownership-examples.md:33]: lint fix:

```suggestion
```

File: docs/standard/native-interop/com-lifetimes-and-ownership-examples.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    int a;
+    float b;
+}
+class PrimitiveClass
+{
+    int a;
+    float b;
+}
+```
+```cpp
```
Comment (BillWagner) [inline docs/standard/native-interop/com-lifetimes-and-ownership-examples.md:52]: add blank line:

```suggestion

```cpp
```

File: docs/standard/native-interop/com-lifetimes-and-ownership-examples.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+int Method2(float* y);
+int Method3(PrimitiveStruct* s);
+struct PrimitiveStruct
+{
+    int a;
+    float b;
+}
+```
+
+
```
Comment (BillWagner) [inline docs/standard/native-interop/com-lifetimes-and-ownership-examples.md:63]: lint:

```suggestion
```

File: docs/standard/native-interop/com-lifetimes-and-ownership-examples.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    PrimitiveClass a;
+    float b;
+}
+class PrimitiveClass
+{
+    int a;
+    float b;
+}
+```
+```cpp
```
Comment (BillWagner) [inline docs/standard/native-interop/com-lifetimes-and-ownership-examples.md:80]: add blank line:

```suggestion

```cpp
```

File: docs/standard/native-interop/com-lifetimes-and-ownership-examples.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    float b;
+}
+struct PrimitiveStruct
+{
+    int a;
+    float b;
+}
+```
+
+
```
Comment (BillWagner) [inline docs/standard/native-interop/com-lifetimes-and-ownership-examples.md:95]: again:

```suggestion
```

File: docs/standard/native-interop/com-lifetimes-and-ownership-examples.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    PrimitiveClass a;
+    float b;
+}
+class PrimitiveClass
+{
+    int a;
+    float b;
+}
+```
+```cpp
```
Comment (BillWagner) [inline docs/standard/native-interop/com-lifetimes-and-ownership-examples.md:112]: and one more:

```suggestion

```cpp
```

File: docs/standard/native-interop/com-lifetimes-and-ownership-examples.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+## Reference to struct with recursive reference
+
+## Array with blittable type
+
+## Array of references to struct
+
+## Array of references to struct with reference to blittable type
+
+## Array of references to struct with recursive reference
```
Comment (BillWagner) [inline docs/standard/native-interop/com-lifetimes-and-ownership-examples.md:135]: This looks unfinished. Are you planning more work here? Should we move the PR into draft status?

**Review by BillWagner — COMMENTED** [review]
Hi @jtschuster 

I left a few comments, but I didn't do a thorough review as it looks like there's more work to do in some of the sections.

File: docs/standard/native-interop/com-lifetimes-and-ownership-examples.md (hunk 8 above)
Comment (jtschuster) [reply docs/standard/native-interop/com-lifetimes-and-ownership-examples.md:135]: Yes, I forgot about this PR. I'll move this to a draft until it's complete. Thanks for reminding me about this!

**Comment by BillWagner** [comment]
@jtschuster 

I'm closing this as very stale. It's been more than a year since the last comment, and it's still in draft status. If you want to reopen the PR, address the conflicts and the comments, we'll take another look.

**Pull request closed without merging**